packages/react-vapor/src/components/table-hoc/TableWithPredicateGeneric.tsx

```diff
@@ -0,0 +1,119 @@
+import {IReactVaporState} from '../../ReactVapor';
+import {ConfigSupplier, HocUtils} from '../../utils/HocUtils';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {UrlUtils} from '../../utils/UrlUtils';
+import {SelectSelector} from '../select/SelectSelector';
+import * as styles from './styles/TableWithPredicates.scss';
+import {ITableHOCOwnProps} from './TableHOC';
+import {TableHOCUtils} from './utils/TableHOCUtils';
+
+export interface IPredicateComponentProps extends Partial<ITableWithPredicateStateProps>,
+ITableHOCOwnProps,
+WithServerSideProcessingProps {
+    id: string;
+}
+
+type PredicateComponentType = React.ComponentType<IPredicateComponentProps>;
+
+export interface ITableWithPredicateGenericConfig extends WithServerSideProcessingProps {
+    id: string;
+    matchPredicate?: (predicate: string, datum: any) => boolean;
+}
+
+export interface ITableWithPredicateStateProps {
+    predicate: string;
+    urlPredicate: string[];
+}
+
+export interface ITableWithPredicateProps
+    extends Partial<ITableWithPredicateStateProps>,
+        ITableHOCOwnProps,
+        WithServerSideProcessingProps {}
+
+const TableWithPredicatePropsToOmit = keys<ITableWithPredicateStateProps>();
+
+const defaultMatchPredicate = (predicate: string, datum: any) =>
+    !predicate || _.some(_.values(datum), (value: string) => value === predicate);
+
+type TableWithPredicateComponent = React.ComponentType<ITableWithPredicateProps>;
+
+export const tableWithPredicateGeneric = (supplier: ConfigSupplier<ITableWithPredicateGenericConfig>) => (
+    PredicateComponent: PredicateComponentType
+) => (
+    TableComponent: TableWithPredicateComponent
+): TableWithPredicateComponent => {
+    const config = HocUtils.supplyConfig(supplier);
+
+    const mapStateToProps = (
+        state: IReactVaporState,
+        ownProps: ITableWithPredicateProps
+    ): ITableWithPredicateStateProps | ITableHOCOwnProps => {
+        const predicate =
+            SelectSelector.getListBoxSelected(state, {
+                id: TableHOCUtils.getPredicateId(ownProps.id, config.id),
+            })[0] || '';
+        const matchPredicate = config.matchPredicate || defaultMatchPredicate;
+        const predicateData = () =>
+            !ownProps.isServer && !config.isServer && predicate
+                ? _.filter(ownProps.data, (datum: any) => matchPredicate(predicate, datum))
+                : ownProps.data;
+
+        const urlParams = UrlUtils.getSearchParams();
+        const possiblePredicates = TableHOCUtils.getPredicateIds(ownProps.id, state);
+
+        return {
+            predicate: predicate,
+            data: ownProps.data && predicateData(),
+            urlPredicate: Object.keys(urlParams)
+                .filter((key) => possiblePredicates.includes(key))
+                .map((key) => urlParams[key]),
+        };
+    };
+
+    @ReduxConnect(mapStateToProps)

```

Comment:
The decoration is deprecated.

You can do something like this instead.

```
const mapStateToProps = (state) => {[...]}

const enhance = connect(mapStateToProps) 

const TableWithPredicateBase: React.FunctionComponent<ITableWithPredicateProps> = () => {}

return enhance(TableWithPredicateBase)

Reply:
As much as those changes would be good, I first wanted to introduce the wrapper with as less changes as possible. 
If you take a look at the diff from my screenshots, the `tableWithPredicateGeneric` is the same old `tableWithPredicate` without the SingleConnect component.

I can make these enhancement in a subsequent pr if needed

Reply:
Fair enough ! :+1: